docs/docs-beta/docs/guides/data-modeling/asset-factories.md

```diff
@@ -1,5 +1,80 @@
 ---
-title: "Creating asset factories"
+title: 'Creating domain-specific languages with asset factories'
 sidebar_position: 60
-sidebar_label: "Creating asset factories"
+sidebar_label: 'Creating domain-specific languages'
 ---
+
+Often times in data engineering, you'll find yourself needing to create a large number of similar assets. For example, you might have a set of tables in a database that all have the same schema, or a set of files in a directory that all have the same format. In these cases, it can be helpful to create a factory that generates these assets for you.
```

Comment:
Lots of good info in here, but it feels like a lot for an intro. Thoughts on something like this?

```
Often in data engineering, you'll find yourself needing to create a large number of similar assets. For example:

- A set of database tables all have the same schema
- A set of files in a directory all have the same format

In some cases, you may be serving stakeholders who aren't familiar with Python or Dagster. They may prefer interacting with assets using a domain-specific language (DSL) built on top of a configuration language such as YAML.

Using an asset factory pattern can solve both of these problems.
```

I didn't include the `In this guide...` sentence in the original as the **What you'll learn** section covers it. However, it also removes the mention of the DSL layer, which we may want to include in here or update the section heading to include it.